server/coin-scraper.js

```diff
@@ -9,25 +9,30 @@ module.exports = class CoinScraper {
     this.previousItemsMap = {};
   }
 
-  fetchMostRecentListedItems() {
-    const requestOptions = {
-      method: 'GET',
-      uri: 'https://pro-api.coinmarketcap.com/v1/cryptocurrency/listings/latest',
-      qs: {
-        'start': '1',
-        'limit': '200',
-        'sort': 'date_added',
-        'sort_dir': 'asc',
-        'aux': 'date_added'
-      },
-      headers: {
-        'X-CMC_PRO_API_KEY': CMC_API_KEY
-      },
-      json: true,
-      gzip: true
-    };
-
-    return rp(requestOptions);
+  async fetchMostRecentListedItems() {
+    try {
+      const requestOptions = {
+        method: 'GET',
+        url: 'https://pro-api.coinmarketcap.com/v1/cryptocurrency/listings/latest',
+        qs: {
+          'start': '1',
+          'limit': '200',
+          'sort': 'date_added',
+          'sort_dir': 'asc',
+          'aux': 'date_added'
+        },
+        headers: {
+          'X-CMC_PRO_API_KEY': CMC_API_KEY
+        },
+        json: true,
+        gzip: true
+      };
+      const response = await axios(requestOptions);
+      // The response coming from api can be accessed through response.data.data that is why we are returning response.data from here
+      return response.data;
+    } catch (error) {
+      return error;
```

Comment:
@Shapokhan how will any method who calls this method `fetchMostRecentListedItems()` find out if the api call returned successfully or it failed (i-e returned with some error) ? The calling method will have to check the structure of the response to find out, right ? Wouldn't it be better to save the calling method from taking that pain by bubbling the exception / error back to it ?

Reply:
To explain what i mean:
```
function xyz() {
  try {
    let res = await fetchMostRecentListedItems();
    // if fetchMostRecentListedItems() never throws an error, i will probably have to examine `res` object to find out myself if the response was success or failure and the thrown an error from here myself so that my catch block is trigerred.
  } catch(err) {
    console.log(err); 
 }
}
```

Reply:
@omeryousaf got your point. I should throw an error instead of returning it.